src/Random.elm

```diff
@@ -522,13 +525,45 @@ you use a seed, you'll get another one back.
 initialSeed : Int -> Seed
 initialSeed x =
     let
-        (Seed state1) =
-            next (Seed 0)
+        -- the default increment magic constant is taken from Numerical Recipes
+        (Seed state1 incr) =
+            next (Seed 0 1013904223)
 
         state2 =
             Bitwise.shiftRightZfBy 0 (state1 + x)
     in
-        next (Seed state2)
+        next (Seed state2 incr)
+
+
+{-| A generator that produces a seed that is independent of any other seed in
+the program. These seeds will generate their own unqiue sequences of random
+values. They are useful when you need an unknown amount of randomness *later*
+but can request only a fixed amount of randomness *now*.
+
+The independent seeds are extremely likely to be distinct for all practical
+purposes. However, it is not proven that there are no pathological cases.
+-}
+independentSeed : Generator Seed
+independentSeed =
+    Generator <|
+        \seed0 ->
+            let
+                gen =
+                    int 0 0xFFFFFFFF
```

Comment:
`Int` is a JavaScript number and the fact that it is a whole number is enforced by the type system, not the representation. If you examine a list of numbers made from this generator, the range is exactly as expected: zero to 2^32 - 1. Note that when a value is passed to `initialSeed`, it undergoes an *unsigned* rightshift which forces it to be positive. But, it appears no such check happens for the increment, so it can be negative. Which, according to the bounds [here](https://en.wikipedia.org/wiki/Linear_congruential_generator), is wrong. So you found a bug!